Implement index_put inplace

manuelcandales opened the pull request:
Differential Revision: D74765433

JacobSzwejbka commented on kernels/portable/cpu/op_index_put.cpp:
```diff
@@ -154,6 +157,175 @@ Tensor& index_put_out(
+
+} // namespace
+
+Tensor& index_put_(
```
is it worth making the out variant call the inplace one on out?

manuelcandales committed:
Implement index_put inplace (#10996)

Summary: Pull Request resolved: https://github.com/pytorch/executorch/pull/10996

Reviewed By: JacobSzwejbka

Differential Revision: D74765433
